src/caffe/proto/caffe.proto

```diff
@@ -443,6 +443,10 @@ message ArgMaxParameter {
   // If true produce pairs (argmax, maxval)
   optional bool out_max_val = 1 [default = false];
   optional uint32 top_k = 2 [default = 1];
+  // The axis along which to maximise -- may be negative to index from the
+  // end (e.g., -1 for the last axis).
+  // By default ArgMaxLayer maximises blobs as flattened array per image.
```

Comment:
Instead of focusing on images, "per instance" or "ArgMaxLayer maximizes over the flattened trailing dimensions for each index of the first / num dimension` might be better since all kinds of predictions could be argmaxed.